File: IN JAVA/StateTaxBracket.java
package com.retireup;

public class StateTaxBracket {
	
	public static float taxBracketArizona(float income, float roi, int year){
		
		float taxes = 0;
		float[] taxbrackets = TaxBeans.gettaxBracketArizona(roi, year);
		float[] tax_rates = TaxBeans.gettax_ratesArizona();
		float[] amounts = TaxBeans.getAmountsArizona(year, roi, tax_rates);
		
		int length = taxbrackets.length;
		
		float[] tax_tax_rate_amount = TaxBracket.calculateTaxBracket(taxbrackets, length, income, amounts, tax_rates);
		
		float taxbracket = tax_tax_rate_amount[0];
		float tax_rate = tax_tax_rate_amount[1];
		float amount = tax_tax_rate_amount[2];
		
		income = income - taxbracket;
		taxes = CalculateTaxes.calculateTaxes(income, tax_rate, amount);
				
		return taxes;
		
	}//end of taxBracketArizona
	
	public static float taxBracketCalifornia(float income, float roi, int year){
		
		float taxes=0;
		float[] taxbrackets = TaxBeans.gettaxBracketCalifornia(roi, year);
		float[] tax_rates = TaxBeans.gettax_ratesCalifornia();
		float[] amounts = TaxBeans.getAmountsCalifornia(year, roi, tax_rates);
		
		int length = taxbrackets.length;
		
		float[] tax_tax_rate_amount = TaxBracket.calculateTaxBracket(taxbrackets, length, income, amounts, tax_rates);
		
		float taxbracket = tax_tax_rate_amount[0];
		float tax_rate = tax_tax_rate_amount[1];
		float amount = tax_tax_rate_amount[2];
		
		//System.out.println(" "+tax_rate+" "+amount);
		income = income - taxbracket;
		taxes = CalculateTaxes.calculateTaxes(income, tax_rate, amount);	
		
		return taxes;
		
	}//end of taxBracketCalifornia
		
	
	public static float taxBracketKansas(float income, float roi, int year){
		
		float taxes = 0;
		float[] taxbrackets = TaxBeans.gettaxBracketKansas(roi, year);
		float[] tax_rates = TaxBeans.gettax_ratesKansas();	
		float[] amounts = TaxBeans.getAmountsKansas(year, roi, tax_rates);	
			
		int length = taxbrackets.length;
		
		float[] tax_tax_rate_amount = TaxBracket.calculateTaxBracket(taxbrackets, length, income, amounts, tax_rates);
		
		float taxbracket = tax_tax_rate_amount[0];
		float tax_rate = tax_tax_rate_amount[1];
		float amount = tax_tax_rate_amount[2];
		
		income = income - taxbracket;
		taxes = CalculateTaxes.calculateTaxes(income, tax_rate, amount);
		
		return taxes;
		
	}//end of taxBracketKansas	
	
		
	
}//end of class

File: Node.js/stateTaxBracket.js
var tax_beans = require('./taxBeans');
var tax_bracket = require('./taxBracket');
var tax_controller = require('./taxController');
module.exports = {	
	
	taxBracketArizona: function(income, roi, year){
		
			//getting current year's taxbrackets for arizona
				var taxbrackets = tax_beans.gettaxBracketArizona(roi, year);
				
			//getting current year's taxrates for arizona
				var tax_rates = tax_beans.gettax_ratesArizona();
			
			//getting current year's amounts for arizona
				var amounts = tax_beans.gettaxAmountsArizona(year, roi, tax_rates);
				
				var length = taxbrackets.length;
				
			//getting taxes
				var taxes = tax_controller(taxbrackets, length, income, amounts, tax_rates);
				
				return taxes;
		
	},
	
	taxBracketCalifornia: function(income, roi, year){
		
		//getting current year's taxbrackets for california
				var taxbrackets = tax_beans.gettaxBracketCalifornia(roi, year);
				
		//getting current year's taxrates for california
				var tax_rates = tax_beans.gettax_ratesCalifornia();
				
		//getting current year's amounts for california
				var amounts = tax_beans.gettaxAmountsCalifornia(year, roi, tax_rates);
				
				var length = taxbrackets.length;
				
		//getting taxes
				var taxes = tax_controller(taxbrackets, length, income, amounts, tax_rates);
				
				return taxes;
		
	},
	
	taxBracketKansas: function(income, roi, year){
		
		//getting current year's taxbrackets for kansas
				var taxbrackets = tax_beans.gettaxBracketKansas(roi, year);
				
		//getting current year's taxrates for kansas
				var tax_rates = tax_beans.gettax_ratesKansas();
				
		//getting current year's amounts for kansas
				var amounts = tax_beans.gettaxAmountsKansas(year, roi, tax_rates);
				
				var length = taxbrackets.length;
				
		//getting taxes
				var taxes = tax_controller(taxbrackets, length, income, amounts, tax_rates);
				
				
				return taxes;
			
	}
};
